docs/usage/type-conversions.md

```diff
@@ -1,114 +1,186 @@
+| `del proxy[foo]`          | `x.delete(foo)`        |
+| `proxy1 == proxy2`        | `x === y`              |
+| `proxy.typeof`            | `typeof x`             |
+| `iter(proxy)`             | `x[Symbol.iterator]()` |
+| `next(proxy)`             | `x.next()`             |
+| `await proxy`             | `await x`              |
+| `proxy.object_entries()`  | `Object.entries(x)`    |
+
+Some other code snippets:
+```python
+for v in proxy:
+    # do something
+```
+is equivalent to:
+```javascript
+for(let v of x){
+    // do something
+}
+```
+The `dir` method has been overloaded to return all keys on the prototype chain
+of `x`, so `dir(x)` roughly translates to:
+```javascript
+function dir(x){
+    let result = [];
+    do {
+        result.push(...Object.getOwnPropertyNames(x));
+    } while (x = Object.getPrototypeOf(x));
+    return result;
+}
+```
+
+As a special case, Javascript `Array`, `HTMLCollection`, and `NodeList` are
+container types, but instead of using `array.get(7)` to get the 7th element,
+javascript uses `array["7"]`. For these cases, we translate:
+
+| Python                    | Javascript             |
+|---------------------------|------------------------|
+| `proxy[idx]`              | `x.toString()`         |
+| `proxy[idx] = val`        | `x.foo`                |
+| `idx in proxy`            | `idx in array`         |
+| `del proxy[idx]`          | `proxy.splice(idx)`    |
+
+
+### Proxying from Python into Javascript
+
+When most Python objects are translated to Javascript a `PyProxy` is produced.
+Fewer operations can be overloaded in Javascript than in Python so some
+operations are more cumbersome on a `PyProxy` than on a `JsProxy`. The following
+operations are currently supported:
+
+| Javascript                 | Python                   |
+|----------------------------|--------------------------|
+| `foo in proxy`             | `hasattr(x, 'foo')`      |
+| `proxy.foo`                | `x.foo`                  |
+| `proxy.foo = bar`          | `x.foo = bar`            |
+| `delete proxy.foo`         | `del x.foo`              |
+| `proxy.ownKeys()`          | `dir(x)`                 |
+| `proxy(...)`               | `x(...)`                 |
+| `proxy.foo(...)`           | `x.foo(...)`             |
+| `proxy.length` or `x.size` | `len(x)`                 |
+| `proxy.has(foo)`           | `foo in x`               |
+| `proxy.get(foo)`           | `x[foo]`                 |
+| `proxy.set(foo, bar)`      | `x[foo] = bar`           |
+| `proxy.delete(foo)`        | `del x[foo]`             |
+| `x.type`                   | `type(x)`                |
+| `x[Symbol.iterator]()`     | `iter(x)`                |
+| `x.next()`                 | `next(x)`                |
+| `await x`                  | `await x`                |
+| `Object.entries(x)`        |  `repr(x)`               |
+
+An additional limitation is that when proxying a Python object into Javascript,
```

Comment:
This could be a warning or a note see [admonitions](https://myst-parser.readthedocs.io/en/latest/using/syntax-optional.html?highlight=admonition#code-fences-using-colons) in myst parser. 

Generally we should maybe use those more in the docs.